Suppress warnings:
  clang diag: warning: unknown warning option '-Wno-unused-but-set-global'; did you mean '-Wno-unused-but-set-variable'? [-Wunknown-warning-option]
  clang diag: warning: unknown warning option '-Wno-unsafe-buffer-usage-in-static-sized-array'; did you mean '-Wno-unsafe-buffer-usage-in-container'? [-Wunknown-warning-option]

diff -up chromium-151.0.7922.137/build/config/compiler/BUILD.gn.clang++-unknown-argument chromium-151.0.7922.137/build/config/compiler/BUILD.gn
--- chromium-151.0.7922.137/build/config/compiler/BUILD.gn.clang++-unknown-argument	2026-08-12 22:51:25.005846860 +0200
+++ chromium-151.0.7922.137/build/config/compiler/BUILD.gn	2026-08-12 22:57:40.710373048 +0200
@@ -576,14 +576,6 @@ config("compiler") {
   if (is_clang) {
     # Flags for diagnostics.
     cflags += [ "-fcolor-diagnostics" ]
-    if (!is_wasm) {
-      if (!is_win) {
-        cflags += [ "-fdiagnostics-show-inlining-chain" ]
-      } else {
-        # Combine after https://github.com/llvm/llvm-project/pull/192241
-        cflags += [ "/clang:-fdiagnostics-show-inlining-chain" ]
-      }
-    }
     if (diagnostics_print_source_range_info && !is_win) {
       cflags += [ "-fdiagnostics-print-source-range-info" ]
     }
@@ -617,9 +609,6 @@ config("compiler") {
     # The performance improvement does not seem worth the risk. See
     # https://crbug.com/484082200 for background and https://crrev.com/c/7593035
     # for discussion.
-    if (!is_wasm) {
-      cflags += [ "-fno-lifetime-dse" ]
-    }
 
     # TODO(hans): Remove this once Clang generates better optimized debug info
     # by default. https://crbug.com/765793
@@ -1563,12 +1552,7 @@ config("clang_warning_suppression") {
 ubsan_hardening("c_array_bounds") {
   sanitizer = "array-bounds"
   condition = !(is_asan && target_cpu == "x86") && !is_wasm
-
-  # Because we've enabled array-bounds sanitizing we also want to suppress
-  # the related warning about "unsafe-buffer-usage-in-static-sized-array",
-  # since we know that the array bounds sanitizing will catch any out-of-
-  # bounds accesses.
-  cflags = [ "-Wno-unsafe-buffer-usage-in-static-sized-array" ]
+  cflags = [ ]
 }
 
 # Enable UBSan subset to ensure we always hit `return` statement in non-void
@@ -1917,9 +1901,6 @@ config("default_warnings") {
       cflags += [
         # TODO(crbug.com/432275627): Fix and re-enable.
         "-Wno-uninitialized-const-pointer",
-
-        # TODO(crbug.com/495753203): Fix and re-enable.
-        "-Wno-unused-but-set-global",
       ]
     }
 
diff -up chromium-151.0.7922.137/build/config/sanitizers/sanitizers.gni.clang++-unknown-argument chromium-151.0.7922.137/build/config/sanitizers/sanitizers.gni
--- chromium-151.0.7922.137/build/config/sanitizers/sanitizers.gni.clang++-unknown-argument	2026-08-11 00:27:27.000000000 +0200
+++ chromium-151.0.7922.137/build/config/sanitizers/sanitizers.gni	2026-08-12 22:51:25.007516573 +0200
@@ -540,7 +540,6 @@ template("ubsan_hardening") {
         # be usable even in release builds, i.e. as widely as possible.
         # It's important not to have full-on UBSan workarounds activate
         # just because we built support for a specific sanitizer.
-        "-fsanitize-ignore-for-ubsan-feature=${invoker.sanitizer}",
       ]
       if (defined(invoker.cflags)) {
         cflags += invoker.cflags
